### PR TITLE
[REF] stock,mrp: clean data,onchange,fields,...

### PR DESCRIPTION
**Pull request by ryv-odoo**
odoo/upgrade#1998

### CODE REVIEW

File: addons/mrp/models/stock_move.py [hunk 1]
```diff
@@ -127,23 +127,10 @@ def _compute_order_finished_lot_ids(self):
...
-                move.allowed_operation_ids = self.env['mrp.routing.workcenter']
-            else:
-                operation_domain = [
-                    ('id', 'in', move.raw_material_production_id.bom_id.operation_ids.ids),
-                    '|',
-                        ('company_id', '=', self.company_id.id),
-                        ('company_id', '=', False)
-                ]
-                move.allowed_operation_ids = self.env['mrp.routing.workcenter'].search(operation_domain)
+            move.allowed_operation_ids = move.raw_material_production_id.bom_id.operation_ids.ids
```
Comment (Whenrow) [inline addons/mrp/models/stock_move.py:133]: ça pose pas de soucis si raw_material_production_id n'est pas défini ?

File: addons/mrp/models/mrp_workorder.py [hunk 2]
```diff
@@ -140,12 +140,7 @@ def _read_group_workcenter_id(self, workcenters, domain, order):
     production_date = fields.Datetime('Production Date', related='production_id.date_planned_start', store=True, readonly=False)
     json_popover = fields.Char('Popover Data JSON', compute='_compute_json_popover')
     show_json_popover = fields.Boolean('Show Popover?', compute='_compute_json_popover')
-    consumption = fields.Selection([
-        ('strict', 'Strict'),
-        ('warning', 'Warning'),
-        ('flexible', 'Flexible')],
-        required=True,
-    )
+    consumption = fields.Selection(related='production_id.consumption')
```
[the remark this answers is not in the document]
Comment (Whenrow) [reply addons/mrp/models/mrp_workorder.py:143]: on ne veux pas que la consumption change en cours de production si tu change sur la bom

File: addons/stock/models/stock_quant.py [hunk 3]
```diff
@@ -656,6 +656,7 @@ class QuantPackage(models.Model):
     _name = "stock.quant.package"
     _description = "Packages"
     _order = 'name'
+    _rec_name = 'name'
```
Comment (Whenrow) [inline addons/stock/models/stock_quant.py:659]: Already done by default
https://github.com/odoo/odoo/blob/a57d4377fb2706e4d21a86cd01e9556b0e2501ba/odoo/models.py#L2808

File: addons/mrp/models/mrp_workorder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryv-odoo) [reply addons/mrp/models/mrp_workorder.py:143]: It is the consumption of the workorder (not the one of the production). And the consumption of the production shouldn't change (as you say) then no problem for the workorder

File: addons/stock/models/stock_quant.py (hunk 3 above)
Comment (ryv-odoo) [reply addons/stock/models/stock_quant.py:659]: Yes, right, it will a better diff

File: addons/mrp/models/stock_move.py (hunk 1 above)
Comment (ryv-odoo) [reply addons/mrp/models/stock_move.py:133]: Not, I don't think so, the implicite mapped will do the job. But the ids at the end can be remove :+1:

File: addons/mrp/models/mrp_bom.py [hunk 4]
```diff
@@ -438,19 +429,10 @@ class MrpByProduct(models.Model):
...
-                byproduct.allowed_operation_ids = self.env['mrp.routing.workcenter']
-            else:
-                operation_domain = [
-                    ('id', 'in', byproduct.bom_id.operation_ids.ids),
-                    '|',
-                        ('company_id', '=', byproduct.company_id.id),
-                        ('company_id', '=', False)
-                ]
-                byproduct.allowed_operation_ids = self.env['mrp.routing.workcenter'].search(operation_domain)
+            byproduct.allowed_operation_ids = byproduct.bom_id.operation_ids.ids
```
Comment (nseinlet) [inline addons/mrp/models/mrp_bom.py:435]: looks like a related

Comment (ryv-odoo) [reply addons/mrp/models/mrp_bom.py:435]: Yes, agree. I just need to change the type of fields in One2many. Thanks, it cleaner.

File: addons/stock/models/stock_move.py [hunk 5]
```diff
@@ -133,7 +133,7 @@ def _default_group_id(self):
         'Propagate cancel and split', default=True,
         help='If checked, when this move is cancelled, cancel the linked move too')
     delay_alert_date = fields.Datetime('Delay Alert Date', help='Process at this date to be on time', compute="_compute_delay_alert_date", store=True)
-    picking_type_id = fields.Many2one('stock.picking.type', 'Operation Type', check_company=True)
+    picking_type_id = fields.Many2one('stock.picking.type', 'Operation Type', compute="_compute_picking_type_id", store=True, check_company=True)
```
Comment (Whenrow) [inline addons/stock/models/stock_move.py:136]: check_company seems useless now regarding the way the field is computed

Comment (ryv-odoo) [reply addons/stock/models/stock_move.py:136]: Well, the picking_type_id is a compute store and we can imagine to set it with something else (where picking_id or production_id is False). By example in the test we set picking_type_id manually. I prefer to keep it by security and the cost is near of zero.

File: addons/stock/models/stock_move.py [hunk 6]
```diff
@@ -874,8 +868,14 @@ def _get_relevant_state_among_moves(self):
...
 
+    @api.onchange('product_id', 'picking_type_id')
+    def _onchange_product(self):
+        if self.product_id:
+            product = self.product_id.with_context(lang=self._get_lang())
+            self.description_picking = product._get_description(self.picking_type_id)
+
     @api.onchange('product_id')
-    def onchange_product_id(self):
+    def _onchange_product_id(self):
```
Comment (Whenrow) [inline addons/stock/models/stock_move.py:878]: why don't you merge this onchange with the one above ?

File: addons/mrp/models/mrp_production.py [hunk 7]
```diff
@@ -642,15 +642,14 @@ def _onchange_location_dest(self):
             }))]
         self.move_finished_ids = update_value_list
 
+
```
Comment (Whenrow) [inline addons/mrp/models/mrp_production.py:645]: :scream:

File: addons/stock/models/stock_move.py (hunk 6 above)
Comment (ryv-odoo) [reply addons/stock/models/stock_move.py:878]: No

File: addons/stock/models/stock_move.py (hunk 5 above)
Comment (amoyaux) [inline addons/stock/models/stock_move.py:136]: single quote

File: addons/stock/models/stock_move.py [hunk 8]
```diff
@@ -157,10 +157,10 @@ def _default_group_id(self):
     warehouse_id = fields.Many2one('stock.warehouse', 'Warehouse', help="Technical field depicting the warehouse to consider for the route selection on the next procurement (if any).")
     has_tracking = fields.Selection(related='product_id.tracking', string='Product with Tracking')
     quantity_done = fields.Float('Quantity Done', compute='_quantity_done_compute', digits='Product Unit of Measure', inverse='_quantity_done_set')
-    show_operations = fields.Boolean(related='picking_id.picking_type_id.show_operations', readonly=False)
+    show_operations = fields.Boolean(related='picking_id.picking_type_id.show_operations', readonly=True)
```
Comment (amoyaux) [inline addons/stock/models/stock_move.py:160]: remove readonly

File: addons/stock/models/stock_picking.py [hunk 9]
```diff
@@ -592,9 +592,12 @@ def onchange_picking_type(self):
             self.location_id = location_id
             self.location_dest_id = location_dest_id
             (self.move_lines | self.move_ids_without_package).update({
-                "picking_type_id": self.picking_type_id,
+                "picking_type_id": self.picking_type_id,  # The compute store doesn't work in case of One2many inverse (move_ids_without_package)
                 "company_id": self.company_id,
             })
+            for move in (self.move_lines | self.move_ids_without_package):
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py:598]: explain this change (comment or other commit)

File: addons/stock/models/stock_move.py [hunk 10]
```diff
@@ -180,12 +180,6 @@ def _default_group_id(self):
     reservation_date = fields.Date('Date to Reserve', compute='_compute_reservation_date', store=True,
```
Comment (amoyaux) [inline addons/stock/models/stock_move.py:180]: "make it prevate" in commit message :1st_place_medal:

File: addons/mrp/views/mrp_bom_views.xml [hunk 11]
```diff
@@ -103,7 +103,7 @@
                                 <field name="operation_ids"
                                     attrs="{'invisible': [('type','not in',('normal','phantom'))]}"
                                     groups="mrp.group_mrp_routings"
-                                    context="{'default_company_id': company_id, 'default_product_tmpl_ids': [product_tmpl_id]}"/>
+                                    context="{'from_bom_form': True, 'default_company_id': company_id, 'default_bom_id': id}"/>
```
Comment (amoyaux) [inline addons/mrp/views/mrp_bom_views.xml:106]: bom_id_invisible?

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**